File: Assets/GameMain/LuaScripts/Procedure/ProcedureMain.lua


local ProcedureMain = class("ProcedureMain",ProcedureBase)

function ProcedureMain:Create()
	ProcedureBase:Create("Main")
end

function ProcedureMain:OnEnter(procedureOwner)
	ProcedureBase:OnEnter(procedureOwner)
	self.procedureOwner = procedureOwner
	print("LuaMainEnter")
end
function ProcedureMain:OnUpdate(procedureOwner, elapseSeconds, realElapseSeconds)
	-- body
end
function ProcedureMain:OnNotify(param0,param1)

end

function ProcedureMain:OnLeave(procedureOwner, isShutdown)
	ProcedureBase:OnLeave(procedureOwner, isShutdown)
end

function ProcedureMain:OnDestroy()
	-- body
end
return ProcedureMain.New()

File: Assets/GameMain/Scripts/Procedure/ProcedureMain.cs
using System.Collections;
using System.Collections.Generic;
using GameFramework;
using GameFramework.Fsm;
using GameFramework.Procedure;
using UnityEngine;
using UnityGameFramework.Runtime;

namespace Game
{
    public class ProcedureMain : ProcedureBase
    {
        private TestLoad m_Load = null;
        public override bool UseNativeDialog
        {
            get
            {
                return false;
            }
        }

        protected internal override void OnEnter(IFsm<IProcedureManager> procedureOwner)
        {
            base.OnEnter(procedureOwner);
            m_Load = new TestLoad();
        }

        protected internal override void OnUpdate(IFsm<IProcedureManager> procedureOwner, float elapseSeconds, float realElapseSeconds)
        {
            base.OnUpdate(procedureOwner, elapseSeconds, realElapseSeconds);
            //测试
            if (Input.GetKeyDown(KeyCode.L))
            {

                m_Load.CreateEntity();
                Debug.LogError(SkillVO.GetConfig(1).Name);
                LuaEventArgs arg = ReferencePool.Acquire<LuaEventArgs>().Fill("Main");
                arg.IntArgs[0] = 1;
                arg.IntArgs[1] = 2;
                GameEntry.Event.Fire(this, arg);
            }
        }

        public override string LuaScriptName
        {
            get
            {
                return "Procedure/ProcedureMain";
            }
        }

    }
}

